### PR TITLE
Add v25.0.0-alpha.1 and Release CRD

### PR DESCRIPTION
**Pull request by nprokopic**
Towards https://github.com/giantswarm/roadmap/issues/3466

### Added

- Add v25.0.0-alpha.1 AWS (CAPA) release with the new cluster-aws version that is being tested.
- Generate Release CRD in the SDK repo from the Release go struct.

### Removed

- Remove v25.0.0-alpha.0 AWS (CAPA) release that was initially added for testing and it now replaced by v25.0.0-alpha.1.

### CODE REVIEW

**Comment by bdehri** [comment]
Couple of differences with vintage:

name: cluster-autoscaler
version: 1.25.3-gs2

name: net-exporter
version: 1.18.2

name: security-bundle
version: 1.6.4

Besides those, I think etcd-kubernetes-resources-count-exporter is the correct name for app, not etcd-k8s-res-count-exporter.

Also, We need to merge this [PR](https://github.com/giantswarm/cluster-aws/pull/636) first to add 4 missing apps.

**Comment by nprokopic** [comment]
> name: cluster-autoscaler
version: 1.25.3-gs2

Ouch, that may be a big diff, let's see if we can downgrade it in v25 🙈 

> I think etcd-kubernetes-resources-count-exporter is the correct name for app, not etcd-k8s-res-count-exporter.

We have new app name in CAPI, IIRC we had to rename it because the old app name was too long.

> Also, We need to merge this https://github.com/giantswarm/cluster-aws/pull/636 first to add 4 missing apps.

Thanks, will add those missing apps here as well!

File: capa/kustomization.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 commonAnnotations:
   giantswarm.io/docs: https://docs.giantswarm.io/ui-api/management-api/crd/releases.release.giantswarm.io/
 resources:
-- v25.0.0-alpha.0
+- v25.0.0-alpha.1
```
Comment (nprokopic) [inline capa/kustomization.yaml:4]: Deleted v25.0.0-alpha.0 as it was just for testing and does not work.

File: sdk/config/crd/replacements/release_name_validation.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: release-name-validation
+  namespace: release-name-validation
+data:
+  pattern: '^(((aws|azure|vsphere|cloud-director|eks)-)|v)(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(-[\.0-9a-zA-Z]*)?$'
```
Comment (nprokopic) [inline sdk/config/crd/replacements/release_name_validation.yaml:7]: Release CR name validation.

P.S. This ConfigMap is just used for name pattern patching when building the Release CRD. The ConfigMap itself is not output to resulting manifests.

File: sdk/manifests/apiextensions.k8s.io_v1_customresourcedefinition_releases.release.giantswarm.io.yaml [hunk 3]
```diff
@@ -0,0 +1,171 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (nprokopic) [inline sdk/manifests/apiextensions.k8s.io_v1_customresourcedefinition_releases.release.giantswarm.io.yaml:1]: Release CRD built by kustomize. It takes Release CRD from `config/crd/bases/release.giantswarm.io_releases.yaml` and applies replacement for updating CR name pattern.

**Review by Gacko — APPROVED** [review]

**Comment by nprokopic** [comment]
> Missing apps:
> 
> name: prometheus-blackbox-exporter version: 0.4.1
> 
> name: k8s-audit-metrics version: 0.9.0
> 
> name: irsa-servicemonitors version: 0.0.1
> 
> name: aws-ebs-csi-driver-servicemonitors version: 0.1.0

Added missing apps to v25.0.0-alpha.1

**Pull request merged**